test/t8n/t8n.cpp

```diff
@@ -117,15 +119,28 @@ int main(int argc, const char* argv[])
 
                         if (loaded_tx_hash_opt != computed_tx_hash)
                             throw std::logic_error("transaction hash mismatched: computed " +
-                                                   hex0x(computed_tx_hash) + ", expected " +
+                                                   computed_tx_hash_str + ", expected " +
                                                    hex0x(loaded_tx_hash_opt.value()));
                     }
 
+                    if (tracing_enabled)
+                    {
+                        // Remove old tracers
+                        vm.set_option("stdtrace", "no");
+                        auto output_filename =
+                            fs::path(output_dir / ("trace-" + std::to_string(i) + "-" +
+                                                      computed_tx_hash_str + ".jsonl"));
+                        // Add 'to file' tracer. Casting required by windows build.
+                        vm.set_option("stdtrace", (const char*)(output_filename.c_str()));
```

Comment:
According to [this](https://cplusplus.com/reference/string/string/c_str/) `c_str` returns `const char*` so no need to typecast?

Reply:
Unfortunately this is required by MSVC https://app.circleci.com/pipelines/github/ethereum/evmone/4313/workflows/9e36645f-85db-4644-b7e3-1efdef3e6553/jobs/46959

Reply:
No, you definitely cannot do it like this. You cast `wchar_t*` to `char* what is totally wrong. `std::path` has a method to convert to `std::string` from "native" string.